util/profile/profile_test.go

```diff
@@ -14,45 +14,43 @@
 package profile_test
 
 import (
+	"testing"
 	"time"
 
-	. "github.com/pingcap/check"
 	"github.com/pingcap/tidb/domain"
 	"github.com/pingcap/tidb/kv"
 	"github.com/pingcap/tidb/session"
 	"github.com/pingcap/tidb/store/mockstore"
+	"github.com/pingcap/tidb/testkit"
 	"github.com/pingcap/tidb/util/profile"
-	"github.com/pingcap/tidb/util/testkit"
+	"github.com/stretchr/testify/require"
 )
 
-type profileSuite struct {
-	store kv.Storage
-	dom   *domain.Domain
-}
+func TestProfiles(t *testing.T) {
+	t.Parallel()
+	var err error
+	var store kv.Storage
+	var dom *domain.Domain
 
-var _ = Suite(&profileSuite{})
+	store, err = mockstore.NewMockStore()
+	require.Nil(t, err)
+	defer func() {
+		err := store.Close()
+		require.Nil(t, err)
+	}()
 
-func (s *profileSuite) SetUpSuite(c *C) {
-	var err error
-	s.store, err = mockstore.NewMockStore()
-	c.Assert(err, IsNil)
 	session.DisableStats4Test()
-	s.dom, err = session.BootstrapSession(s.store)
-	c.Assert(err, IsNil)
-}
-
-func (s *profileSuite) TearDownSuite(c *C) {
-	s.dom.Close()
-	s.store.Close()
-}
+	dom, err = session.BootstrapSession(store)
+	require.Nil(t, err)
+	defer dom.Close()
```

Comment:
Maybe put this line in the defer of store, and require nil.

Reply:
This method never returns err, and I don't know what you mean by "put this line in the defer of store". Could you share a patch?

Reply:
OK, I forgot that It did not return an error. Resolved.

Reply:
Resolved as invalid. If you can provide a patch, feel free to reopen this one.